### PR TITLE
Corsify errors

### PR DESCRIPTION
**Pull request by mgreystone**
@flintinatux 's much better fix for #21 

Add CORS headers for error responses (status code >= 400)

### CODE REVIEW

File: lib/cors.js [hunk 1]
```diff
@@ -12,6 +12,10 @@ const basics = opts => ({
   'access-control-allow-origin': opts.origin || defs.origin
 })
 
+const corsifyError = opts => err => {
+  throw Object.assign(err, { headers: basics(opts) })
```
Comment (mgreystone) [inline lib/cors.js]: I did think of this but was not confident that we were cool with this not being immutable.

Comment (mgreystone) [reply lib/cors.js]: > Oh, and using Ramda to merge results in an object that isn't an Error, which has it's own class of problems.

Yeah, i noticed that 😢

**Comment by pklingem** [comment]
@mgreystone have you tried your business against this?

**Comment by mgreystone** [comment]
Yes, this works for what i need.

**Comment by pklingem** [comment]
coors squirrel

**Pull request merged**